geoapps/drivers/components/locations.py

```diff
@@ -0,0 +1,83 @@
+#  Copyright (c) 2021 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+from uuid import UUID
+
+import numpy as np
+from geoh5py.objects import Grid2D
+
+from geoapps.utils import rotate_xy
+
+
+class InversionLocations:
```

Comment:
This is a great idea. See comment above to consolidate more of the core xyz operations.